lib/livebook/runtime/erl_dist/runtime_server.ex

```diff
@@ -477,13 +478,14 @@ defmodule Livebook.Runtime.ErlDist.RuntimeServer do
   end
 
   def handle_cast({:put_system_envs, envs}, state) do
-    System.put_env(envs)
+    envs |> normalize_path_env(state) |> System.put_env()
 
     {:noreply, state}
   end
 
   def handle_cast({:delete_system_envs, names}, state) do
     Enum.each(names, &System.delete_env/1)
+    System.put_env("PATH", state.initial_path)
```

Comment:
```suggestion
    Enum.each(names, fn
      "PATH" -> System.put_env("PATH", state.initial_path)
      name -> System.delete_env(name)
    end)
```

Reply:
Makes more sense, to not triggering the PATH change everytime